core/domain/exp_domain.py

```diff
@@ -565,6 +566,22 @@ def from_dict(
                 state_domain.Solution.from_dict(idict['id'], idict['solution'])
                 if idict['solution'] else None)
 
+            if idict['id']:
```

Comment:
This seems like quite a fundamental change. Should it be here?

Reply:
It does need to be there, previously, if cust args were missing (which occurs when we are migrating from v1, for example), it was handled in InteractionInstance (see [here](https://github.com/oppia/oppia/blob/79047c1cdb5b08dab3f2d6d8b1c09c52d042d223/core/domain/state_domain.py#L458)). However, now that adding cust args also adds content_id's, the content_id's need to be added to written_translations and recorded_voiceovers, and it needs to happen before InteractionInstance.to_dict() for validation purposes. The change here does the same as before- populates missing cust args but also adds it to written_translations and recorded_voiceovers.

Reply:
Now that I think about it, this could go in the state migration instead. Does that sound good?